confluent_kafka/schema_registry/json_schema.py

```diff
@@ -0,0 +1,294 @@
+                                           Serializer)
+
+
+class _ContextStringIO(BytesIO):
+    """
+    Wrapper to allow use of StringIO via 'with' constructs.
+
+    """
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+        return False
+
+
+class JsonSerializer(Serializer):
+    """
+    JsonSerializer serializes objects in the Confluent Schema Registry binary
+    format for JSON.
+
+    JsonSerializer configuration properties:
+    +-----------------------+----------+--------------------------------------------------+
+    | Property Name         | Type     | Description                                      |
+    +=======================+==========+==================================================+
+    |                       |          | Registers schemas automatically if not           |
+    | auto.register.schemas | bool     | previously associated with a particular subject. |
+    |                       |          | Defaults to True.                                |
+    +-----------------------|----------+--------------------------------------------------+
+    |                       |          | Callable(SerializationContext, str) -> str       |
+    |                       |          |                                                  |
+    | subject.name.strategy | callable | Instructs the JsonSerializer on how to construct |
+    |                       |          | Schema Registry subject names.                   |
+    |                       |          | Defaults to topic_subject_name_strategy.         |
+    +-----------------------+----------+--------------------------------------------------+
+
+    Schemas are registered to namespaces known as Subjects which define how a
+    schema may evolve over time. By default the subject name is formed by
+    concatenating the topic name with the message field separated by a hyphen.
+
+    i.e. {topic name}-{message field}
+
+    Alternative naming strategies may be configured with the property
+    `subject.name.strategy`.
+
+    Supported subject name strategies:
+    +--------------------------------------+------------------------------+
+    | Subject Name Strategy                | Output Format                |
+    +======================================+==============================+
+    | topic_subject_name_strategy(default) | {topic name}-{message field} |
+    +--------------------------------------+------------------------------+
+    | topic_record_subject_name_strategy   | {topic name}-{record name}   |
+    +--------------------------------------+------------------------------+
+    | record_subject_name_strategy         | {record name}                |
+    +--------------------------------------+------------------------------+
+
+    See ``Subject name strategy`` for additional details.
+
+    Note:
+        The ``title`` annotation, referred to as a record name
```

Comment:
this needs rewording a bit i think to be precisely correct, but that can wait.

Reply:
sure thing. I actually want to do a docs specific PR to verify/correct any rendering problems, wording etc.